src/reducers/user.js

```diff
@@ -1,29 +1,55 @@
-const mockState = {
-  isAuth: true,
-  user_id: 1,
+const initialState = {
+    isAuth: false,
+    user_id: 0,
+    userEvents: [],
+    userGroups: {
+        owned: [],
+        joined: []
+    }
 };
 
-const userReducer = (state = {}, action) => {
-  switch (action.type) {
-    case "LOGIN":
-      localStorage.setItem("user", {
-        isAuth: true,
-        user_id: action.payload.user_id,
-      });
-      return {
-        ...state,
-        isAuth: true,
-        user_id: action.payload.user_id,
-      };
-    case "LOGOUT":
-      localStorage.removeItem("user")
-      return {
-        ...state,
-        isAuth: false,
-        user_id: null,
-      };
-    default:
-      return state;
-  }
+const userReducer = (state = initialState, action) => {
+    switch (action.type) {
+        case "LOGIN":
+            localStorage.setItem("user", {
+                isAuth: true,
+                user_id: action.payload.user_id,
+            });
+            return {
+                ...state,
+                isAuth: true,
+                user_id: action.payload.user_id,
+            };
+        case "LOGOUT":
+            localStorage.removeItem("user")
+            return {
+                ...state,
+                isAuth: false,
+                user_id: null,
+            };
+        case "ADD_GROUP": {
+            return {
+                ...state,
+                userGroups: {
+                    ...state.userGroups.owned,
```

Comment:
isnt it better to just use `...state.userGroups,`?

Reply:
what do you mean?